src/components/ContactsAdd.js

```diff
@@ -1,39 +1,144 @@
-import { useState } from "react"
-import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+
+const initialData = {
+  firstName: "",
+  lastName: "",
+  street: "",
+  city: "",
+  email: "",
+  truthSocial: "",
+  linkedIn: "",
+};
 
 function ContactsAdd(props) {
+  const contacts = props.contacts;
+  const setContacts = props.setContacts;
+  const [inputData, setInputData] = useState(initialData);
+  const navigate = useNavigate();
+
+  const handleChange = (ev) => {
+    const { name, value, type } = ev.target;
+
+    if (name === "firstName" && type === "text") {
+      setInputData({ ...inputData, firstName: value });
+    }
+    if (name === "lastName" && type === "text") {
+      setInputData({ ...inputData, lastName: value });
+    }
+    if (name === "street" && type === "text") {
+      setInputData({ ...inputData, street: value });
+    }
+    if (name === "city" && type === "text") {
+      setInputData({ ...inputData, city: value });
+    }
+    if (name === "email" && type === "email") {
+      setInputData({ ...inputData, email: value });
+    }
+    if (name === "linkedIn" && type === "text") {
+      setInputData({ ...inputData, linkedIn: value });
+    }
+    if (name === "truthSocial" && type === "text") {
+      setInputData({ ...inputData, truthSocial: value });
+    }
+  };
```

Comment:
Nice use of the spread operator.
To clean up even more you can use the 'name' variable as a key, like this:

`setInputData({ ...inputData, [name]: value });`

This replaces all if statements in one line.